src/rgw/rgw_appmain.cc

```diff
@@ -581,13 +584,25 @@ void rgw::AppMain::init_lua()
   }
 } /* init_lua */
 
+void rgw::AppMain::init_dedup()
+{
+  rgw::sal::Driver* driver = env.driver;
+  if (driver->get_name() == "rados") { /* Supported for only RadosStore */
+    dedup_background = std::make_unique<rgw::dedup::Background>(driver, dpp->get_cct());
```

Comment:
this can throw an exception in the cluster ctor.
but would be difficult for the calling code here to know that.
probably a better approach is not to do anything that can fail in the cluster ctor, and instead add an `init()` function to both the `Background` class and the `cluster` class